mypy/checker.py

```diff
@@ -1452,6 +1460,55 @@ def check_simple_assignment(self, lvalue_type: Type, rvalue: Expression,
                                    '{} has type'.format(lvalue_name))
             return rvalue_type
 
+    def check_descriptor_assignment(self, instance_type: Type, descriptor_type: Type,
+                                    rvalue: Expression, context: Context) -> Tuple[Type, bool]:
+        """
+        Type check descriptor assignment.
+
+        Return the inferred rvalue_type and whether to infer anything about the lvalue type
+        """
+        # Descriptors don't participate in class-attribute access
+        if isinstance(instance_type, FunctionLike) and instance_type.is_type_obj():
+            rvalue_type = self.check_simple_assignment(descriptor_type, rvalue, context)
```

Comment:
I couldn't come up with a reasonable way to do that, since we need both a type and whether to infer anything about the result type. I re-named the function to try and capture its details more completely.